sycl/doc/extensions/DeviceGlobal/SYCL_INTEL_device_global.asciidoc

```diff
@@ -0,0 +1,774 @@
+  
+  static_assert(std::is_trivially_default_constructible_v<T>,
+                "Type T must be trivially default constructable (until C++20 "
+                "consteval is supported and enabled)");
+
+  static_assert(std::is_trivially_destructible_v<T>,
+      "Type T must be trivially destructible.");
+
+  // Only default construction is allowed.  The underlying memory allocations
+  // of type T on devices will be zero-initialized before any non-initialization
+  // accesses occur.
+  device_global();
+
+  device_global(const device_global &) = delete;
+  device_global(const device_global &&) = delete;
+  device_global &operator=(const device_global &) = delete;
+  device_global &operator=(const device_global &&) = delete;
+
+  template <access::decorated IsDecorated>
+  multi_ptr<T, access::address_space::global_space, IsDecorated> get_multi_ptr() noexcept;
+  template <access::decorated IsDecorated>
+  multi_ptr<const T, access::address_space::global_space, IsDecorated> get_multi_ptr() const noexcept;
+
+  // Access the underlying data
+  operator T&() noexcept;
+  operator const T&() const noexcept;
+ 
+  T& get() noexcept;
+  const T& get() const noexcept;
+
+  // Enable assignments from underlying type
+  device_global& operator=(const T&) noexcept;
+
+  // Available if the operator[] is valid for objects of type T
+  using subscript_return_t = std::remove_reference_t<decltype(std::declval<T>()[std::ptrdiff_t{}])>;
+  subscript_return_t& operator[]( std::ptrdiff_t idx ) noexcept;
+  const subscript_return_t& operator[]( std::ptrdiff_t idx ) const noexcept;
+
+  // Available if the operator-> is valid for objects of type T
+  T& operator->() noexcept;
+  const T& operator->() const noexcept;
+
+  // Note that there is no need for "device_global" to define member functions for
+  // operators like "++", comparison, etc.  Instead, the type "T" need only define
+  // these operators as non-member functions.  Because there is an implicit conversion
+  // from "device_global" to "T&", the operations can be applied to objects of type
+  // "device_global<T>".
+
+  template<typename propertyT>
+  static constexpr bool has_property();
+
+  // The return type is an unspecified internal class used to represent 
+  // instances of propertyT
+  template<typename propertyT>
+  static constexpr auto get_property();
+};
+
+} // namespace sycl::ext::oneapi
+----
+
+[frame="topbot",options="header"]
+|===
+|Functions |Description
+
+// --- ROW BREAK ---
+a|
+[source,c++]
+----
+device_global();
+----
+|
+Constructs a `device_global` object, and implicit storage for `T` in the global address space on each device that may access it.
+
+The storage on each device for `T` is zero-initialized.
+
+`T` must be trivially default constructable and trivially destructible.
+
+// --- ROW BREAK ---
+a|
+[source,c++]
+----
+template <access::decorated IsDecorated>
+multi_ptr<T, access::address_space::global_space, IsDecorated> get_multi_ptr() noexcept;
```

Comment:
```suggestion
multi_ptr<T, access::address_space::global_space, IsDecorated>
  get_multi_ptr() noexcept;
```

The HTML render for this table causes the contents of the right column to be cut off.  I suspect this is because the left column it too wide.  I suspect it may help to split these long lines as I show above.  Same for the long declaration below.

Reply:
Done in 3e04b444abe7b8600880790d41c976cdad7d351c